python/activator/middleware_interface.py

```diff
@@ -776,6 +777,37 @@ def ingest_image(self, oid: str) -> None:
         assert len(result) == 1, "Should have ingested exactly one image."
         _log.info("Ingested one %s with dataId=%s", result[0].datasetType.name, result[0].dataId)
 
+    def _get_graph_executor(self, butler, factory):
+        """Create a QuantumGraphExecutor suitable for Prompt Processing.
+
+        Parameters
+        ----------
+        butler : `lsst.daf.butler.Butler`
+            The Butler for which the quantum graph will be generated
+            and executed. Should match the Butler passed to
+            SeparablePipelineExecutor.
+        factory : `lsst.pipe.base.TaskFactory`
+            The task factory used for pipeline execution. Should match
+            the factory passed to SeparablePipelineExecutor.
+
+        Returns
+        -------
+        executor : `lsst.ctrl.mpexec.QuantumGraphExecutor`
+            The executor to use.
+        """
+        quantum_executor = SingleQuantumExecutor(
+            butler,
+            factory,
+        )
+        graph_executor = MPGraphExecutor(
+            # TODO: re-enable parallel execution once we can log as desired with CliLog or a successor
+            # (see issues linked from DM-42063)
```

Comment:
Looks like a good workaround for the time being. Maybe we should create a new ticket for the future work discussed at Slack on more properly engineering this even if that may not happen soon?

Reply:
I think it will need to be done piece by piece, was hoping to discuss the details with Middleware after catching up on backlog.